mathjax3-ts/output/chtml/Wrappers/mpadded.ts

```diff
@@ -0,0 +1,138 @@
+/*************************************************************
+ *
+ *  Copyright (c) 2017 The MathJax Consortium
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+/**
+ * @fileoverview  Implements the CHTMLmpadded wrapper for the MmlMpadded object
+ *
+ * @author dpvc@mathjax.org (Davide Cervone)
+ */
+
+import {CHTMLWrapper, StringMap} from '../Wrapper.js';
+import {BBox} from '../BBox.js';
+import {MmlMpadded} from '../../../core/MmlTree/MmlNodes/mpadded.js';
+import {MmlNode} from '../../../core/MmlTree/MmlNode.js';
+import {Property} from '../../../core/Tree/Node.js';
+
+/*
+ * Needed in order to access BBox properties by variable
+ */
+interface IBBox extends BBox {
```

Comment:
BBox is a class, so IBBox should not extend it.
Interfaces should extend other interfaces and classes should implement interfaces.
You should probably have an interface declaration in BBox.ts that BBox implements. 
Maybe for the time being it's also sufficient to declare IBBox as a type, e.g.:
export type IBBox [name: string]: number | string | Function;

Reply:
I worked out how to get around needing `IBBox`.  It was just to be able to access a `BBox`'s properties via a variable, `bbox[x]`, with would throw an error because `BBox` has no index structure.  But I can use `bbox[x as (keyof BBox)]` and that works.